mcs/tools/al/Al.cs

```diff
@@ -594,42 +594,48 @@ private void SetKeyPair (AssemblyName aname)
 			if (isTemplateFile) {
 				// LAMESPEC: according to MSDN, the template assembly must have a
 				// strong name but this is not enforced
-				Assembly assembly = Assembly.LoadFrom (templateFile);
+				var asmDef = Mono.Cecil.AssemblyDefinition.ReadAssembly (templateFile);
 
 				// inherit signing related settings from template, but do not
 				// override command-line options
-				object [] attrs = assembly.GetCustomAttributes (true);
-				foreach (object o in attrs) {
-					if (o is AssemblyKeyFileAttribute) {
+				foreach (var attr in asmDef.CustomAttributes) {
```

Comment:
I tested this with the assembly having a custom attribute from an assembly which was not available when `al` was run, and it didn't crash. This is the exact use case that I am trying to fix here in fact.